### ISSUE TITLE
Failed registration process still enters email address in database

### ISSUE THREAD

**Issue by PropGit** [parallaxinc/BlocklyProp#122]
It's possible for a registration process to end up in a failed state, yet still enters the account's email address into the database; preventing a repeated attempt from completing because "Email already exists."

We need to find a way to resolve this; either through transaction-based processing, or some other solution that prevents a user from feeling stuck.

In the case I experienced most recently, clicking on the Register button left the browser spinning endlessly- no error and no timeout before I gave up and stopped it.  There was no indication that it finished any part of the process; yet, trying again was immediately met with an "Email already exists" error.

**Comment by PropGit** [comment]
I can't easily verify that this is resolved or not.  I will close it and if I see it occur again, I will reopen the issue.

**Comment by zfi** [comment]
This likely happens if the SMTP configuration is broken. The email send attempt happens after the initial customer account data is put in the database. The email send fails so the customer is unable to confirm the registration. 

This should be reproducible on a dev VM that has no SMTP server configured.

**Comment by PropGit** [comment]
Thanks @zfi - I tested it in the Development VM and didn't get a failed state response; it actually said it successfully sent the email, though of course it could not.  The response I got this time (as opposed to what started this issue) is recoverable, assuming a user clicks the Email not yet confirmed link and is able to get an email the second time around.

I'm leaving this issue closed as I can not verify that the nature of the problem still exists.
